extension-dev.code-workspace

```diff
@@ -197,42 +286,129 @@
           "searchPaths": [],
           "searchMicrosoftSymbolServer": true,
           "searchNuGetOrgSymbolServer": true
+        },
+        "presentation": {
+          "hidden": false,
+          "group": "test",
+          "order": 3
         }
       },
       {
-        "name": "Launch Extension",
+        // Runs the extension in an empty temp profile that is automatically cleaned up after use
+        // Undocumented: https://github.com/microsoft/vscode-docs/issues/6220
+        "name": "Launch Extension - Temp Profile",
         "type": "extensionHost",
         "request": "launch",
         "runtimeExecutable": "${execPath}",
         "args": [
-          "--disable-extensions",
-          "--extensionDevelopmentPath=${workspaceFolder:Client}"
+          "--profile-temp",
+          "--extensionDevelopmentPath=${workspaceFolder:Client}",
+          "${workspaceFolder:Client}/examples"
         ],
         "sourceMaps": true,
+        // This speeds up source map detection and makes smartStep work correctly
         "outFiles": [
-          "${workspaceFolder:Client}/out/main.js"
+          "${workspaceFolder:Client}/out/**/*.js",
+          "!**/node_modules/**",
+          "!**/.vscode-test/**"
+        ],
+        "skipFiles": [
+          "<node_internals>/**",
+          "**/node_modules/**",
+          "**/.vscode-test/**"
         ],
-        "preLaunchTask": "${defaultBuildTask}",
+        "presentation": {
+          "hidden": false,
+          "group": "test",
+          "order": 2
+        }
       },
       {
-        "name": "Launch Extension Tests",
+        // Runs the extension in an isolated but persistent profile separate from the user settings
+        // Undocumented: https://github.com/microsoft/vscode-docs/issues/6220
+        "name": "Launch Extension - Isolated Profile",
```

Comment:
What's the use case of this versus the temporary profile?

Reply:
It's persistent. Temp profile is thrown away each time, in this one you can test some persistent stuff interactively, like lets say you want to manually set up some settings to test something, it'll persist and then you can just delete it after.

Reply:
Ah I see!